### ISSUE TITLE
FlashCards: Index arguments provided by user

### ISSUE THREAD

**Issue by huiminlim** [AY1920S1-CS2103-F09-2/main#109]
tldr: users will enter index starting from 1,2,3,... when accessing/modifying list elements.

Currently, there is a inconsistency in usage of the "index" arguments provided by the user to refer to the respective components of the flashcards, i.e. choices, cards in deck etc. This problem exists in `FlashCard`, `Deck`, `State`

I plan to align to the usage to the following:

User-input: 

1-based indexing strictly, expect an index >= 1 each time a user enters any index to refer to any item in FlashCard, be it choices, card in deck etc.
An exception will be thrown if this is violated.

In the methods:

Will reduce the index by 1, then proceed with checks to check validity (not exceeding number of cards available)
Any index used by the methods will be 0-based.

The various arguments/attribute names will be updated to reflect this slight difference.

Do let me know if any concerns, apologies for this discrepancy!
